lib/addons/s3-csi-driver/index.ts

```diff
@@ -0,0 +1,85 @@
+import { Construct } from "constructs";
+import { ClusterInfo } from "../../spi";
+import { HelmAddOn, HelmAddOnUserProps } from "../helm-addon";
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { createNamespace, setPath, supportsALL } from "../../utils";
+import { getS3DriverPolicyStatements } from "./iam-policy";
+
+const S3_CSI_DRIVER_SA = 's3-csi-driver-sa';
+const S3_CSI_DRIVER = 's3-csi-driver';
+const S3_CSI_DRIVER_RELEASE = 's3-csi-driver-release';
+const S3_DRIVER_POLICY = 's3-csi-driver-policy';
+
+/**
+ * Configuration options for the add-on.
+ */
+export interface S3CSIDriverAddOnProps extends HelmAddOnUserProps {
+    /**
+     * ARN of the S3 bucket to be used by the driver
+     */
+    s3BucketName: string;
+    /**
+     * Create Namespace with the provided one (will not if namespace is kube-system)
+     */
+    createNamespace?: boolean
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: HelmAddOnUserProps & S3CSIDriverAddOnProps = {
+    chart: S3_CSI_DRIVER,
+    name: S3_CSI_DRIVER,
+    namespace: 'kube-system',
+    release: S3_CSI_DRIVER_RELEASE,
+    version: 'v1.9.0',
+    repository: 'https://github.com/awslabs/mountpoint-s3-csi-driver',
```

Comment:
This should be 'https://awslabs.github.io/mountpoint-s3-csi-driver' which is a proper helm repo as opposed to the source repo. Otherwise helm version validation fails. 
See https://github.com/awslabs/mountpoint-s3-csi-driver/blob/main/docs/install.md#helm